src/calliope/config.py

```diff
@@ -82,27 +82,20 @@ def update(self, update_dict: dict, deep: bool = False) -> Self:
         self._kwargs = update_dict
         return updated
 
-    @overload
-    def model_yaml_schema(self, filepath: str | Path) -> None: ...
-
-    @overload
-    def model_yaml_schema(self, filepath: None = None) -> str: ...
-
-    def model_yaml_schema(self, filepath: str | Path | None = None) -> None | str:
-        """Generate a YAML schema for the class.
+    def model_json_schema(self, replace_refs=False) -> AttrDict:
```

Comment:
I would _not_ override a parent method. We can just remove this entirely and have a utility function that converts the json schema output to a referenceless schema.